implementations/rust/kex/src/xx.rs

```diff
@@ -550,7 +577,17 @@ mod tests {
         let res = state.prologue();
         assert!(res.is_ok());
```

Comment:
@mikelodder7 @SanjoDeundiak, there seem to be a lot of panic-inducing calls throughout this code `assert` `assert_eq` `unwrap` etc.. 

We need to either prove their existence is acceptable, or handle the cases in which they fail. I'm not confident that we're ok with these here, but would be happy to know otherwise 😄  ultimately, we should never panic and take down a caller's process unless there is no other logical option. We let them make that decision after they handle our returned error.

Reply:
@nilslice is that so important for test code?

Reply:
@SanjoDeundiak -- ah, no tests are definitely a good place to use those kinds of calls. I apologize if any of my comments in the review are associated with tests. Github doesn't do a great job showing the context of the code point... 

All tests should use those kinds of functions, and we should be very selective when we use them elsewhere.